addons/mail/static/src/components/message/message.xml

```diff
@@ -91,7 +91,7 @@
                             <t t-if="message.moderation_status === 'pending_moderation' and !message.isModeratedByCurrentPartner">
                                 <span class="o_Message_moderationPending o-author" title="Your message is pending moderation.">Pending moderation</span>
                             </t>
-                            <t t-if="message.notifications.length > 0">
+                            <t t-if="(!threadViewer or !message.originThread or message.originThread === threadViewer.thread) and message.notifications.length > 0">
```

Comment:
The point was to be able to do this:
```suggestion
                            <t t-if="threadViewer and message.originThread and message.originThread === threadViewer.thread and message.notifications.length > 0">
```
Which is: we only show failures if we are sure to be on the origin thread, and we don't show it if the origin can't be tested.

Reply:
I guess it will be annoying in some tests though if we do what I said, and since in practice we never have a message without threadViewer or originThread... I'm not sure now.

Reply:
I think mail notification failures have flaws, in that they are related to message but it lacks context (like on which thread it has been posted and/or on which thread(s) the recipient received).

The purpose of PR is to hide them in non-origin thread because it's buggy / not properly working in other threads. As a quick fix, I would enforce limiting this feature on thread viewer on origin thread of message.

Tests related to that becomes more specific, but that's the intent of this quick FIX in the 1st place, so I'm good with that. The _ideal_ fix is much more complex, because it's a lacking aspect in the server modeling that seems to require significant effort, so all in all this is the best option right now.

Reply:
Yes @seb-odoo that's what I've done initially but a lot of tests was red so ... At the end do I have to change something thus ?

Reply:
I don't really care, according to last message from Alex yes we want to change the condition.

Then you have to ensure "createMessageComponent" is passing a thread viewer (matching origin thread) as props to Message.